File: 0-99/56. Merge Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        int size = intervals.size();
        if(size==0 || size==1)
            return intervals;
        sort(intervals.begin(),intervals.end());
        int i=0;
        vector<vector<int>> result;
        vector<int> temp = intervals[0];
        for(int i=1; i<size; i++){
            if(temp[1]>=intervals[i][0])
                temp[1] = max(intervals[i][1], temp[1]);
            else{
                result.push_back(temp);
                temp = intervals[i];
            }
        }
        result.push_back(temp);
        return result;
    }
};

2019.07.29

File: 2025/56. Merge Intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals = sorted(intervals, key=lambda x: x[0])
        
        i = 0
        while i + 1 < len(intervals):
            if intervals[i][1] >= intervals[i + 1][0]:
                intervals[i][1] = max(intervals[i][1], intervals[i + 1][1])
                intervals.pop(i + 1)
            else:
                i += 1
        return intervals
